### PR TITLE
fix: nightly updater

### PR DESCRIPTION
**Pull request by vansangpfiev**
## Describe Your Changes

- bugfix for nightly updater. We can not delete executable file after `cortex update`, we rename it to `cortex_temp` then need to delete at the next start.

## Fixes Issues

- Closes #
- Closes #

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

File: engine/commands/cortex_upd_cmd.h [hunk 1]
```diff
@@ -83,7 +85,8 @@ inline bool ReplaceBinaryInflight(const std::filesystem::path& src,
     // Already has the newest
     return true;
   }
-  std::filesystem::path temp = std::filesystem::temp_directory_path() / "cortex_temp";
+  std::filesystem::path temp =
+      file_manager_utils::GetExecutableFolderContainerPath() / "cortex_temp";
```
Comment (vansangpfiev) [inline engine/commands/cortex_upd_cmd.h]: Should rename in the same folder

File: engine/main.cc [hunk 2]
```diff
@@ -133,6 +133,17 @@ void ForkProcess() {
 int main(int argc, char* argv[]) {
   { file_manager_utils::CreateConfigFileIfNotExist(); }
 
+  // Delete temporary file if it exists
+  auto temp =
+      file_manager_utils::GetExecutableFolderContainerPath() / "cortex_temp";
```
Comment (vansangpfiev) [inline engine/main.cc:140]: After cortex update, we can only remove temporary file at the next start

File: engine/utils/file_manager_utils.h [hunk 3]
```diff
@@ -230,7 +230,7 @@ inline std::filesystem::path GetContainerFolderPath(
   } else if (type == "CudaToolkit") {
     container_folder_path = current_path;
   } else if (type == "Cortex") {
-    container_folder_path = current_path / "cortex";
+    container_folder_path = std::filesystem::temp_directory_path() / "cortex";
```
Comment (vansangpfiev) [inline engine/utils/file_manager_utils.h:233]: Download binary to temp folder

File: engine/commands/cortex_upd_cmd.cc [hunk 4]
```diff
@@ -190,7 +190,8 @@ bool CortexUpdCmd::GetNightly(const std::string& v) {
 
   // Replace binay file
   auto executable_path = file_manager_utils::GetExecutableFolderContainerPath();
-  auto src = executable_path / "cortex" / GetCortexBinary();
+  auto src = std::filesystem::temp_directory_path() / "cortex" / kCortexBinary /
```
Comment (vansangpfiev) [inline engine/commands/cortex_upd_cmd.cc:194]: Change accordingly to s3 structure

**Review by namchuai — APPROVED** [review]
✅  looks good to me

**Pull request merged**